elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/QueryDsl.scala

```diff
@@ -368,4 +370,64 @@ trait QueryDsl extends DslCommons with SortDsl {
       _dis_max -> innerMap
     )
   }
+
+  case class HighlightRoot(queryRoot: QueryRoot, highlight: Highlight)
```

Comment:
Can you rename it to `QueryRootLike`? That's the scala standard way for saying something is a trait.